src/storage_tests.cc

```diff
@@ -0,0 +1,417 @@
+#include <gtest/gtest.h>
+
+#include "storage_tests.h"
+
+namespace storage {
+
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+void StorageTest::SetUp() {
+  src_facility_ = new Storage(tc_.get());
+  InitParameters();
+  SetUpStorage();
+}
+
+void StorageTest::TearDown() {
+  delete src_facility_;
+}
+
+void StorageTest::InitParameters(){
+  in_c1 = "in_c1";
+  out_c1 = "out_c1";
+  in_r1 = "in_r1";
+  out_r1 = "out_r1";
+  process_time = 10;
+  max_inv_size = 200;
+  capacity = 20;
+  cost = 1;
+
+  cyclus::CompMap v;
+  v[922350000] = 1;
+  v[922380000] = 2;
+  cyclus::Composition::Ptr recipe = cyclus::Composition::CreateFromAtom(v);
+  tc_.get()->AddRecipe(in_r1, recipe);
+
+  v[94239] = 0.25;
+  recipe = cyclus::Composition::CreateFromAtom(v);
+  tc_.get()->AddRecipe(out_r1, recipe);
+}
+
+void StorageTest::SetUpStorage(){
+  src_facility_->in_commod_(in_c1);
+  src_facility_->out_commod_(out_c1);
+  src_facility_->in_recipe_(in_r1);
+  src_facility_->out_recipe_(out_r1);
+  src_facility_->process_time_(process_time);
+  src_facility_->max_inv_size_(max_inv_size);
+  src_facility_->capacity_(capacity);
+  src_facility_->cost_(cost);
+}
+
+void StorageTest::TestInitState(Storage* fac){
+  EXPECT_EQ(process_time, fac->process_time_());
+  EXPECT_EQ(max_inv_size, fac->max_inv_size_());
+  EXPECT_EQ(capacity, fac->capacity_());
+  EXPECT_EQ(out_c1, fac->out_commod_());
+  EXPECT_EQ(out_r1, fac->out_recipe_());
+  EXPECT_EQ(in_c1, fac->in_commod_());
+  EXPECT_EQ(in_r1, fac->in_recipe_());
+  EXPECT_EQ(cost, fac->cost_());
+}
+
+void StorageTest::TestRequest(Storage* fac, double cap){
+  cyclus::Material::Ptr req = fac->Request_();
+  EXPECT_EQ(cap, req->quantity());
+}
+
+void StorageTest::TestAddMat(Storage* fac, 
+    cyclus::Material::Ptr mat){
+  double amt = mat->quantity();
+  double before = fac->inventory.quantity();
+  fac->AddMat_(mat);
+  double after = fac->inventory.quantity();
+  EXPECT_EQ(amt, after - before);
+}
+
+void StorageTest::TestBuffers(Storage* fac, double inv, 
+    double proc, double stocks){
+  double t = tc_.get()->time();
+
+  EXPECT_EQ(inv, fac->inventory.quantity());
+  EXPECT_EQ(proc, fac->processing[t].quantity());
+  EXPECT_EQ(stocks, fac->stocks.quantity());
+}
+
+void StorageTest::TestStocks(Storage* fac, cyclus::CompMap v){
+
+  cyclus::toolkit::ResourceBuff* buffer = &fac->stocks;
+  Material::Ptr final_mat = cyclus::ResCast<Material>(buffer->Pop(cyclus::toolkit::ResourceBuff::BACK));
+  cyclus::CompMap final_comp = final_mat->comp()->atom();
+  EXPECT_EQ(final_comp,v);
+
+}
+
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+TEST_F(StorageTest, clone) {
+  Storage* cloned_fac =
+      dynamic_cast<Storage*> (src_facility_->Clone());
+  TestInitState(cloned_fac);
+}
+
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+TEST_F(StorageTest, InitialState) {
+  // Test things about the initial state of the facility here
```

Comment:
Should actually do some testing here. Initial state includes all initialized variables, capacities, expected function behaviors at the start of the simulation, etc.